transport/src/main/java/io/netty/channel/SliceableFileRegion.java

```diff
@@ -0,0 +1,131 @@
+
+    /**
+     * Returns {@code true} if and only if {@code (this.count - this.transferIndex)} is greater than
+     * {@code 0}.
+     */
+    boolean isTransferable();
+
+    /**
+     * Returns the number of readable bytes which is equal to {@code (count - transferIndex)}.
+     */
+    long transferableBytes();
+
+    /**
+     * Transfers the content of this file region to the specified channel.
+     *
+     * @param target
+     *            the destination of the transfer
+     * @param length
+     *            the maximum number of bytes to transfer
+     */
+    long transferBytesTo(WritableByteChannel target, long length) throws IOException;
+
+    /**
+     * Transfers the content of this file region to the specified channel.
+     * <p>
+     * This method does not modify {@link #transferIndex()}.
+     *
+     * @param target
+     *            the destination of the transfer
+     * @param position
+     *            the relative offset of the file where the transfer begins from. For example,
+     *            <tt>0</tt> will make the transfer start from {@link #position()}th byte and
+     *            <tt>{@link #count()} - 1</tt> will make the last byte of the region transferred.
+     * @param length
+     *            the maximum number of bytes to transfer
+     */
+    long transferBytesTo(WritableByteChannel target, long position, long length) throws IOException;
+
+    /**
+     * Returns a slice of this file region's sub-region. This method does not modify
+     * {@code transferIndex} of this buffer.
+     * <p>
+     * Also be aware that this method will NOT call {@link #retain()} and so the reference count
+     * will NOT be increased.
+     */
+    SliceableFileRegion slice(long index, long length);
+
+    /**
+     * Returns a slice of this file region's sub-region. This method does not modify
+     * {@code transferIndex} of this buffer.
+     * <p>
+     * Note that this method returns a {@linkplain #retain() retained} file region unlike
+     * {@link #slice()}.
+     */
+    SliceableFileRegion retainedSlice(long index, long length);
+
+    /**
+     * Returns a new slice of this file region's sub-region starting at the current
+     * {@code transferIndex} and increases the {@code transferIndex} by the size of the new slice (=
+     * {@code length}).
+     * <p>
+     * Also be aware that this method will NOT call {@link #retain()} and so the reference count
+     * will NOT be increased.
+     *
+     * @param length
+     *            the size of the new slice
+     */
+    SliceableFileRegion transferSlice(long length);
+
+    /**
+     * Returns a new slice of this file region's sub-region starting at the current
+     * {@code transferIndex} and increases the {@code transferIndex} by the size of the new slice (=
+     * {@code length}).
+     * <p>
+     * Note that this method returns a {@linkplain #retain() retained} file region unlike
+     * {@link #transferSlice()}.
```

Comment:
this link is dead -> `{@link #transferSlice(long)}.`

Reply:
Done.